src/lib/get-locale.ts

```diff
@@ -0,0 +1,55 @@
+import type { CountryCode } from '@guardian/libs';
+import { getCookie, isString, storage } from '@guardian/libs';
+import type { Edition } from '../types';
+
+const KEY = 'GU_geo_country';
+const KEY_OVERRIDE = 'gu.geo.override';
+const COUNTRY_REGEX = /^[A-Z]{2}$/;
+
+// best guess that we have a valid code, without actually shipping the entire list
+const isValidCountryCode = (country: unknown) =>
```

Comment:
Could we make this a type guard? Might be able to avoid the `as CountryCode` assertion on L46 this way.

```suggestion
const isValidCountryCode = (country: unknown): country is CountryCode =>
```

Reply:
Yup, have changed